.github/workflows/_rust_tests.yml

```diff
@@ -37,7 +37,7 @@ jobs:
   test:
     name: Test rust crates
     if: (!cancelled())
-    runs-on: [self-hosted]
+    runs-on: [self-hosted] # Run on either x64 or arm64
```

Comment:
What makes this special? Also, how would this interact with #5491?

Reply:
Nothing, but it might be a good idea to use the arm64 runners for the rust crates tests if they are idle.
Regarding #5491 I don't now, that is a different topic tbh.
This one is just updating the current jobs to match the correct runners

Reply:
I guess I'm just asking whether you are okay with all of the tests running using a single runner kind, since that will likely be the outcome after merging this PR with that one.

Reply:
The arm runners are new, we have them since two days. So setting everything to "self-hosted-x64" is the same situation as it always was. The arm runners should be primarily for building the binaries for arm, so we don't have to cross compile.